js/exporter/jspdf/v3/draw_utils.js

```diff
@@ -151,7 +153,20 @@ function setLinesStyles(doc, { borderColor }, docStyles) {
     doc.setLineWidth(defaultBorderLineWidth);
     const currentBorderColor = isDefined(borderColor) ? borderColor : docStyles.borderColor;
     if(currentBorderColor !== doc.getDrawColor()) {
-        doc.setDrawColor(currentBorderColor);
+        callMethodWithColorParameter(doc, 'setDrawColor', currentBorderColor);
+    }
+}
+
+function callMethodWithColorParameter(doc, method, color) {
+    if(!isObject(color)) {
+        doc[method](color);
+    } else {
+        const argsCount = Object.keys(color).length;
+        if(argsCount === 3) {
+            doc[method](color.ch1, color.ch2, color.ch3);
+        } else if(argsCount === 4) {
+            doc[method](color.ch1, color.ch2, color.ch3, color.ch4);
+        }
```

Comment:
imho we need to throw an error here:

        const argsCount = Object.keys(color).length;
        if(argsCount === 3) {
            doc[method](color.ch1, color.ch2, color.ch3);
        } else if(argsCount === 4) {
            doc[method](color.ch1, color.ch2, color.ch3, color.ch4);
        } else {
            throw Error(`An incorrect color object was passed: 3 or 4 members are expected while the passed object has ${argsCount} members`);
        }

Reply:
Done